src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -186,6 +186,16 @@ private async Task ProcessGetBlocksAsync(INetworkPeer peer, GetBlocksPayload get
         {
             this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(peer), peer.RemoteSocketEndpoint, nameof(getBlocksPayload), getBlocksPayload);
 
+            if (getBlocksPayload.BlockLocators.Blocks.Count > BlockLocator.MaxLocatorSize)
+            {
+                this.logger.LogTrace("Peer '{0}' sent getblocks with oversized locator, disconnecting.", peer.RemoteSocketEndpoint);
+
+                peer.Disconnect("Peer sent getblocks with oversized locator");
```

Comment:
Ah hang on should we not ban the node?

Reply:
I think core only disconnect from the node:

https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L2013

Reply:
Ah I think its so they dont ban old nodes that may provide locator information differently (BitcoinJ I think) so perhaps we don't care about such cases.
But maybe its so trivial so its not worth adding an entry in the db.

Reply:
Ah I see @Aprogiena commented on this not adding the banning dependency yeah ok lets push this as is (same as core)